databuilder/__main__.py

```diff
@@ -81,21 +100,23 @@ def build_parser(environ):
 def add_generate_dataset(subparsers, environ):
     parser = subparsers.add_parser("generate-dataset", help="Generate a dataset")
     parser.set_defaults(which="generate-dataset")
-    parser.add_argument(
-        "--dataset-definition",
-        help="The path of the file where the dataset is defined",
-        type=existing_python_file,
-    )
     parser.add_argument(
         "--output",
-        help="Path and filename (or pattern) of the file(s) where the dataset will be written",
+        help="Path of the file where the dataset will be written (console by default)",
         type=Path,
     )
+    parser.add_argument(
+        "--dsn",
+        help="Data Source Name: URL of remote database, or path to data on disk",
+        type=str,
+        default=environ.get("DATABASE_URL"),
```

Comment:
question: Should this environment variable name be changed for consistency with the fact it might not be a URL?

Reply:
If we change it here, maybe we could be consistent with the other variable names and use an `OPENSAFELY` prefix. That would:

* avoid risk of collision with any other software — should that software similarly use generic names
* makes it obvious what the environment variable is for
* makes it easier to remember, instead of a special case

Reply:
I deliberately didn't make that change because it will require coordination with job-runner and there's a [bigger question](https://github.com/opensafely-core/databuilder/issues/580#issuecomment-1211754690) there about whether we should use `OPENSAFELY_` or `DATABUILDER_` as our env var prefix that we ought to settle first.

`DATABASE_URL` is something of a de-facto standard so you could frame this as a risk of collision, or as having things work nicely out of the box :smile:

Maybe the thing to do is close #580 and make a new issue to consider all our env var names together?

Reply:
Or maybe name them `DATABUILDER_…` even, as [in @evansd' comment elsewhere](https://github.com/opensafely-core/databuilder/issues/580#issue-1287217034).

Reply:
It doesn't need to block this PR in any case, again, as this PR has some useful usability improvements.